src/processor/operator/copy_from/read_csv.cpp

```diff
@@ -0,0 +1,22 @@
+#include "processor/operator/copy_from/read_csv.h"
+
+using namespace kuzu::storage;
+
+namespace kuzu {
+namespace processor {
+bool ReadCSV::getNextTuplesInternal(ExecutionContext* context) {
+    auto morsel = sharedState->getMorselSerial();
```

Comment:
You can pass in data chunk in resultSet through `getMorselSerial ` here, so you don't need to initialize it for each call inside the function. And should also getting rid of `readDataChunk` function.